internal/pool/pool.go

```diff
@@ -107,16 +105,21 @@ func (p *ConnPool) checkMinIdleConns() {
 	if p.opt.MinIdleConns == 0 {
 		return
 	}
-	if p.poolSize < p.opt.PoolSize && p.idleConnsLen < p.opt.MinIdleConns {
-		p.poolSize++
-		p.idleConnsLen++
-		go p.addIdleConn()
+	if p.poolSize < p.opt.PoolSize {
+		balanceLen := p.opt.MinIdleConns - p.idleConnsLen
+		for i := 0; i < balanceLen; i++ {
+			p.poolSize++
+			p.idleConnsLen++
+			go p.addIdleConn()
+		}
 	}
 }
 
 func (p *ConnPool) addIdleConn() {
 	cn, err := p.newConn(context.TODO(), true)
 	if err != nil {
+		p.poolSize--
+		p.idleConnsLen--
```

Comment:
I believe you should hold `connsMu` here. Otherwise it looks good :+1:

Reply:
Thanks for your review.I updated the code.